docs/source/reference/cpp.rst

```diff
@@ -48,10 +48,65 @@ Math utilities (namespace :cpp:class:`rascal::math`)
     :project: rascal
     :members:
 
+Gradient tests (developer)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. cpp:namespace:: rascal
+
+All representations in libRascal should implement, or plan to implement,
+gradients (derivatives w.r.t. the Cartesian positions of all the atoms) so that
+they can be used to run dynamics.  This is often a complex and error-prone task,
+so a finite-difference gradient checker is provided to check the gradients of
+any representation calculator -- or any mathematical function in general -- and
+ensure that the analytical and finite-difference gradients match up.
+
+To check the gradient of a new representation calculator, it should suffice to
+use the classes RepresentationCalculatorGradientProvider (to provide the function
+and its gradient) and RepresentationCalculatorGradientFixture (to assist in
+iterating over the atoms of the structure).  An example of its usage is shown
+below, excerpted from :file:`tests/test_calculator.cc`:
+
+.. literalinclude:: ../../../tests/test_calculator.cc
+   :language: cpp
+   :lines: 430-436, 441-444
+   :dedent: 4
```

Comment:
you might want to use the following keywords instead of lines to avoid shifts in the future ? 
```
:start-after: my-start
:end-before: my-end
```
You can insert my-start into the source code at the proper line  like this and the same for my-end
```
/* -------------------- my-start -------------------- */
````

Reply:
Ok, it clutters the source code a bit but I suppose that's tolerable.